File: Licenta/Semester 5/PPD/Laboratory4/src/Matrix.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Matrix {
    private List<List<Integer>> content;
    private int rows, cols;

    public Matrix(int rows, int cols) {
       this.rows = rows;
       this.cols = cols;

       int max = 900;
       int min = 100;

       Random rand = new Random();

       this.content = new ArrayList<>(rows);
       for (int i = 0; i < this.rows; i++){
           this.content.add(new ArrayList<>(cols));
           for (int j = 0; j < this.cols; j++){
               this.content.get(i).add(rand.nextInt((max - min) + 1) + min);
           }
       }
    }

    public void FillWithZero() {
        for (int i = 0; i < this.rows; i++) {
            this.content.add(new ArrayList<>(cols));
            for (int j = 0; j < this.cols; j++) {
                this.content.get(i).set(j, 0);
            }
        }
    }

    public int get(int row, int col){
        return this.content.get(row).get(col);
    }

    public void set(int row, int col, int value){
        this.content.get(row).set(col, value);
    }

    public int getRowsNumber() {
        return rows;
    }

    public int getColsNumber() {
        return cols;
    }



    @Override
    public String toString() {
        StringBuilder ss = new StringBuilder();
        for (int i = 0; i < this.rows; i++){
            ss.append(this.content.get(i).toString()).append("\n");
        }

        return ss.toString();
    }
}

File: Licenta/Semester 5/PPD/Laboratory3/Thread Poll C++/Thread Pool/Thread Pool/Matrix.cpp
#include "Matrix.h"
#include <string>	

Matrix::Matrix(int rows, int cols)
{
	this->rows = rows;
	this->cols = cols;
	int max = 900;
	int min = 100;

	for (int i = 0; i < this->rows; i++)
	{
		std::vector<int> vec{};
		this->content.push_back(vec);
		for (int j = 0; j < this->cols; j++)
		{
			int random = rand() % 10 + 1;
			this->content.at(i).push_back(j);
		}
	}
}

Matrix::~Matrix()
{
}


std::string Matrix::printMatrix()
{
	std::string out = "";

	for (int i = 0; i < this->rows; i++)
	{
		for (int j = 0; j < this->cols; j++)
		{
			out += std::to_string(this->content[i][j]) + ' ';
		}
		out += '\n';
	}

	return out;
}

int Matrix::index(int row, int col)
{
	return row * this->cols + col;
}

void Matrix::set(int row, int col, int value)
{
	this->content[row][col] = value;
}

int Matrix::get(int row, int col)
{
	return this->content[row][col];
}

int Matrix::getRowsNumber()
{
	return this->rows;
}

int Matrix::getColsNumber()
{
	return this->cols;
}